src/lib/widget/BufFlatten.svelte

```diff
@@ -0,0 +1,176 @@
+    
+    numFrames = endFrame - startFrame;
+    numChans = endChan - startChan;
+  }
+  
+</script>
+
+<form>
+    <label class='numberboxlabel' for='start_frame'>Start Frame</label>
+    <input class='numberbox' id='start_frame' bind:value={startFrame} type=number min=0 max={srcFrames-1} />
+</form>
+
+<form>
+    <label class='numberboxlabel' for='num_frames'>Num Frames</label>
+    <input class='numberbox' id='num_frames' bind:value={displayNumFrames} type=number min=-1 max={srcFrames-startFrame} />
+    <label class="error">{ numFrames_error }</label>
+</form>
+
+<form>
+    <label class='numberboxlabel' for='start_chan'>Start Chan</label>
+    <input class='numberbox' id='start_chan' bind:value={startChan} type=number min=0 max={srcChans-1}/>
+</form>
+
+<form>
+    <label class='numberboxlabel' for='num_chans'>Num Chans</label>
+    <input class='numberbox' id='num_chans' bind:value={displayNumChans} type=number min=-1 max={srcChans-startChan}/>
+    <label class="error">{ numChans_error }</label>
+</form>
+
+<form>
+    <label class='numberboxlabel' for='axis'>Axis</label>
+    <input class='numberbox' id='axis' bind:value={axis} type=number min=0 max=1/>
+</form>
+
+<br>
+
+<div class="container source">
+    <div class="label">Source Buffer</div>
+    {#each {length: srcChans} as _,chan}
+    <div class="row">
+        {#each {length: srcFrames} as _,frame}
+        <div
+        class='cube'
+        style:background-color={css_colors[frame]} 
+        style:clip-path={shapes[chan]}
+        />
+        {/each}
+    </div>
+    {/each}
+</div>
+
+<div class="container destination">
+    <div class="label">Destination Buffer</div>
+    <div class="row">
+    
+    {#if axis===1}
+        {#each {length: numFrames} as _,frame }
+          {#each {length: numChans} as _,chan }
+            <div 
+            class="cube"
+            style:background-color={css_colors[startFrame+frame]} 
+            style:clip-path={shapes[startChan+chan]}
+            />
+          {/each}                
+        {/each}
+        
+        {:else}
+        
+        {#each {length: numChans} as _,chan }
+          {#each {length: numFrames} as _,frame }
+            <div 
+            class="cube"
+            style:background-color={css_colors[startFrame+frame]} 
+            style:clip-path={shapes[startChan+chan]}
+            />
+          {/each}                
+        {/each}
+         
+        {/if}
+    </div>
+</div>
+
+<style>
+    .numberbox {
+        width: 50px;
+    }
+    
+    .numberboxlabel {
+        width: 150px;
+    }
+    
+    .error {
+        color: red;
+        font-size: 0.75rem;
+    }
+    
+    .container {
+        display: flex;
+        flex-direction: column;
+        gap: 0.5em;
+        margin-bottom: 2em;
+    }
+
+    .row {
+        display: flex;
+        flex-direction: row;
+        gap: 0.5em;
+    }
+
+    .cube {
+        width: 30px;
+        height: 30px;
+    }
+
+    form {
```

Comment:
There are potentially three things in the grid, so we should account for them otherwise layouts will get wonky

Reply:
same as above: I don't know what most of these css attributes actually do, so you'll have to break this down for me.I don't know what most of these css attributes actually do, so you'll have to break this down for me.

Reply:
> same as above: I don't know what most of these css attributes actually do, so you'll have to break this down for me.I don't know what most of these css attributes actually do, so you'll have to break this down for me.

All good. `display: grid` arranges child elements in a grid. the `grid-template` indicates the structure of that grid. We could make it three elements to be explicit about the structure.

Reply:
So we can do:

`10ch auto auto` possibly. I would have to check.